test/language_command_interface_tests.cpp

```diff
@@ -0,0 +1,161 @@
+#include <gtest/gtest.h>
+
+#include "isobus/isobus/can_NAME_filter.hpp"
+#include "isobus/isobus/can_internal_control_function.hpp"
+#include "isobus/isobus/can_managed_message.hpp"
+#include "isobus/isobus/can_parameter_group_number_request_protocol.hpp"
+#include "isobus/isobus/can_partnered_control_function.hpp"
+#include "isobus/isobus/can_stack_logger.hpp"
+#include "isobus/isobus/isobus_language_command_interface.hpp"
+
+using namespace isobus;
+
+// A log sink for the CAN stack
+class CustomLogger : public isobus::CANStackLogger
+{
+public:
+	void sink_CAN_stack_log(CANStackLogger::LoggingLevel, const std::string &) override
+	{
+		// Do nothing
+	}
+};
+
+static CustomLogger logger;
+
+TEST(LANGUAGE_COMMAND_INTERFACE_TESTS, BasicConstructionAndInit)
+{
+	NAME clientNAME(0);
+	auto internalECU = std::make_shared<InternalControlFunction>(clientNAME, 0x26, 0);
+	LanguageCommandInterface interfaceUnderTest(internalECU);
+	EXPECT_EQ(false, interfaceUnderTest.get_initialized());
+	EXPECT_EQ(false, interfaceUnderTest.send_request_language_command());
+	interfaceUnderTest.initialize();
+	EXPECT_EQ(true, interfaceUnderTest.get_initialized());
+	EXPECT_EQ(0, interfaceUnderTest.get_language_command_timestamp());
+	interfaceUnderTest.initialize(); // double init
+	EXPECT_EQ(true, interfaceUnderTest.get_initialized());
+	interfaceUnderTest.send_request_language_command();
+}
+
+TEST(LANGUAGE_COMMAND_INTERFACE_TESTS, InvalidICF)
+{
+	LanguageCommandInterface interfaceUnderTest(nullptr);
+	interfaceUnderTest.initialize();
+	interfaceUnderTest.send_request_language_command();
+}
+
+TEST(LANGUAGE_COMMAND_INTERFACE_TESTS, ValidPartner)
+{
+	std::vector<isobus::NAMEFilter> vtNameFilters;
+	const isobus::NAMEFilter testFilter(isobus::NAME::NAMEParameters::FunctionCode, static_cast<std::uint8_t>(isobus::NAME::Function::VirtualTerminal));
+	vtNameFilters.push_back(testFilter);
+
+	NAME clientNAME(0);
+	auto internalECU = std::make_shared<InternalControlFunction>(clientNAME, 0x26, 0);
+
+	auto vtPartner = std::make_shared<PartneredControlFunction>(0, vtNameFilters);
+	LanguageCommandInterface interfaceUnderTest(internalECU, vtPartner);
+	ParameterGroupNumberRequestProtocol::assign_pgn_request_protocol_to_internal_control_function(internalECU); // Pre-assign our ICF
```

Comment:
It's not strictly required, I was trying to cover a missing condition in the initialize with this, but it's not super valuable.

Reply:
Ohh I see, it's the condition where an parameter control function is already assigned right? I think we should divert this to it's own test file when we implement it later. Since we are now only 'covering' it but not actually testing/'asserting' it.